### ISSUE TITLE
Stable 3.2 `mon_use_fqdn` is undefined error

### ISSUE THREAD

**Issue by noahdesu** [ceph/ceph-ansible#3550]
Trying to run `shrink-osd.yml`. Managed to fix #3538 with

```diff
diff --git a/tests/functional/centos/7/shrink_osd/group_vars/osds b/tests/functional/centos/7/shrink_osd/group_vars/osds
index 04c810da..d07a317f 100644
--- a/tests/functional/centos/7/shrink_osd/group_vars/osds
+++ b/tests/functional/centos/7/shrink_osd/group_vars/osds
@@ -2,6 +2,8 @@
 os_tuning_params:
   - { name: fs.file-max, value: 26234859 }
 osd_scenario: lvm
+journal_size: 100
+osd_objectstore: "filestore"
 lvm_volumes:
   - data: data-lv1
     journal: /dev/sdc1
@@ -9,4 +11,4 @@ lvm_volumes:
   - data: data-lv2
     journal: journal1
     data_vg: test_group
-    journal_vg: journals
\ No newline at end of file
+    journal_vg: journals
```

But then ran into

```
TASK [ceph-facts : set_fact monitor_name ansible_hostname] *******************************************************************************************************************************************************************************************************************************
task path: /home/nwatkins/src/ceph-ansible/roles/ceph-facts/tasks/facts.yml:11                                                                                                                                                                                                            
Tuesday 29 January 2019  13:17:31 -0800 (0:00:00.063)       0:00:02.323 *******                                                                                                                                                                                                           
fatal: [localhost]: FAILED! => {}                                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                          
MSG:                                                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                          
The conditional check 'not mon_use_fqdn' failed. The error was: error while evaluating conditional (not mon_use_fqdn): 'mon_use_fqdn' is undefined                                                                                                                                        
                                                                                                                                                                                                                                                                                          
The error appears to have been in '/home/nwatkins/src/ceph-ansible/roles/ceph-facts/tasks/facts.yml': line 11, column 3, but may                                                                                                                                                          
be elsewhere in the file depending on the exact syntax problem.                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                          
The offending line appears to be:                                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                          
- name: set_fact monitor_name ansible_hostname                                                                                                                                                                                                                                            
  ^ here                                                                                                                              
```

**Comment by noahdesu** [comment]
There are shrink-osd BZs that need to be backported, and being able to run tests is critical.

**Comment by guits** [comment]
@noahdesu mon_use_fqdn has a default value in role `ceph-defaults`
didn't have a chance to look further to your issue but perhaps a call to `ceph-defaults` is missing somewhere in `shrink-osd.yml` ?

**Comment by noahdesu** [comment]
@guits seems to be there https://github.com/ceph/ceph-ansible/blob/stable-3.2/infrastructure-playbooks/shrink-osd.yml#L59

**Comment by guits** [comment]
@noahdesu I must be missing something here, the output you linked indicates you hit this error in `roles/ceph-facts/tasks/facts.yml:11` but there's no call to `ceph-facts` in `shrink-osd.yml`

I think we are missing some info/context in this issue.

**Comment by noahdesu** [comment]
@guits i'll give that a shot

**Comment by noahdesu** [comment]
seems to have been fixed in stable-3.2 with test infra cleanup merged recnetly
